apps/extension/src/shared/components/grpc-endpoint-form/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { FormEvent, useRef } from 'react';
+import { SelectList } from '@penumbra-zone/ui/components/ui/select-list';
+import { Button } from '@penumbra-zone/ui/components/ui/button';
+import { Network } from 'lucide-react';
+import { useGrpcEndpointForm } from './use-grpc-endpoint-form';
+import { cn } from '@penumbra-zone/ui/lib/utils';
+
+/**
+ * Renders all the parts of the gRPC endpoint form that are shared between the
+ * onboarding flow and the RPC settings page.
+ */
+export const GrpcEndpointForm = ({
+  submitButtonLabel,
+  onSuccess,
+}: {
+  submitButtonLabel: string;
+  onSuccess: () => void | Promise<void>;
+}) => {
+  const {
+    chainId,
+    grpcEndpoints,
+    grpcEndpointInput,
+    setGrpcEndpointInput,
+    onSubmit,
+    rpcError,
+    isSubmitButtonEnabled,
+    isCustomGrpcEndpoint,
+  } = useGrpcEndpointForm();
+  const customGrpcEndpointInput = useRef<HTMLInputElement | null>(null);
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (isSubmitButtonEnabled) void onSubmit(onSuccess);
+  };
+
+  return (
+    <div className='flex flex-col gap-2'>
```

Comment:
minor but i'd make this a bare form element instead of wrapping it in a div

Reply:
Yeah I was going to, but I actually have a single element _outside_ the form at the bottom of this component that needs to be spaced apart from the form with `gap-2.